semtypes/src/main/java/io/ballerina/types/definition/ObjectDefinition.java

```diff
@@ -0,0 +1,101 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.types.definition;
+
+import io.ballerina.types.BasicTypeCode;
+import io.ballerina.types.Bdd;
+import io.ballerina.types.CellSemType;
+import io.ballerina.types.Core;
+import io.ballerina.types.Definition;
+import io.ballerina.types.Env;
+import io.ballerina.types.PredefinedType;
+import io.ballerina.types.SemType;
+import io.ballerina.types.SubtypeData;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import static io.ballerina.types.Core.createBasicSemType;
+import static io.ballerina.types.Core.union;
+import static io.ballerina.types.subtypedata.CellSubtype.cellContaining;
+
+/**
+ * Represent object type desc.
+ *
+ * @since 2201.10.0
+ */
+public final class ObjectDefinition implements Definition {
+
+    private final MappingDefinition mappingDefinition = new MappingDefinition();
+
+    public SemType define(Env env, ObjectQualifiers qualifiers, List<Member> members) {
+        Stream<CellField> memberStream = members.stream().map(member -> memberField(env, member));
```

Comment:
```suggestion
        if (members.isEmpty()) {
            return PredefinedType.OBJECT;
        }
        Stream<CellField> memberStream = members.stream().map(member -> memberField(env, member));
```

Reply:
Even if we have no members qualifiers may be not default (ex `client object {}`). I felt it is not worth it to special case checking for qualifiers as well since users are quite unlikely to use top level type anyway giving us any benefits from using a singleton. WDYT?

Reply:
+1 to not have it.